src/particle.rs

```diff
@@ -0,0 +1,692 @@
+                             VibrationEntity particle packet",
+                        )?;
+                        let ticks: VarInt = Decode::decode(r).context(
+                            "Failed to read field `ticks` while decoding a VibrationEntity \
+                             particle packet",
+                        )?;
+                        Particle::VibrationEntity {
+                            entity_id: entity_id.0,
+                            entity_eye_height,
+                            ticks: ticks.0,
+                        }
+                    }
+                    _ => {
+                        bail!("Invalid position_source_type");
+                    }
+                }
+            }
+            41 => Particle::ItemSlime,
+            42 => Particle::ItemSnowball,
+            43 => Particle::LargeSmoke,
+            44 => Particle::Lava,
+            45 => Particle::Mycelium,
+            46 => Particle::Note,
+            47 => Particle::Poof,
+            48 => Particle::Portal,
+            49 => Particle::Rain,
+            50 => Particle::Smoke,
+            51 => Particle::Sneeze,
+            52 => Particle::Spit,
+            53 => Particle::SquidInk,
+            54 => Particle::SweepAttack,
+            55 => Particle::TotemOfUndying,
+            56 => Particle::Underwater,
+            57 => Particle::Splash,
+            58 => Particle::Witch,
+            59 => Particle::BubblePop,
+            60 => Particle::CurrentDown,
+            61 => Particle::BubbleColumnUp,
+            62 => Particle::Nautilus,
+            63 => Particle::Dolphin,
+            64 => Particle::CampfireCosySmoke,
+            65 => Particle::CampfireSignalSmoke,
+            66 => Particle::DrippingHoney,
+            67 => Particle::FallingHoney,
+            68 => Particle::LandingHoney,
+            69 => Particle::FallingNectar,
+            70 => Particle::FallingSporeBlossom,
+            71 => Particle::Ash,
+            72 => Particle::CrimsonSpore,
+            73 => Particle::WarpedSpore,
+            74 => Particle::SporeBlossomAir,
+            75 => Particle::DrippingObsidianTear,
+            76 => Particle::FallingObsidianTear,
+            77 => Particle::LandingObsidianTear,
+            78 => Particle::ReversePortal,
+            79 => Particle::WhiteAsh,
+            80 => Particle::SmallFlame,
+            81 => Particle::Snowflake,
+            82 => Particle::DrippingDripstoneLava,
+            83 => Particle::FallingDripstoneLava,
+            84 => Particle::DrippingDripstoneWater,
+            85 => Particle::FallingDripstoneWater,
+            86 => Particle::GlowSquidInk,
+            87 => Particle::Glow,
+            88 => Particle::WaxOn,
+            89 => Particle::WaxOff,
+            90 => Particle::ElectricSpark,
+            91 => Particle::Scrape,
+            _ => {
+                bail!("Invalid particle id");
+            }
```

Comment:
Would be nice to include the invalid particle ID in the error message like
```suggestion
            id => bail!("invalid particle ID {id}"),
```
Error messages start with lowercase letters by convention.